core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -594,12 +606,24 @@ private[spark] class AppStatusListener(
 
       stage.executorSummaries.values.foreach(update(_, now))
       update(stage, now, last = true)
+
+      val executorIdsForStage = stage.executorSummaries.keySet
+      executorIdsForStage.foreach { executorId =>
+        liveExecutors.get(executorId).foreach { exec =>
+          removeBlackListedStageFrom(exec, event.stageInfo.stageId, now)
```

Comment:
I'm just doing a really quick scan here, but I don't understand why changes here are necessary.  You don't get these events for blacklisting within a stage.

or is this a bug in the current code, and something we should fix in the pending 2.3 release?

Reply:
I guess github diff collapse tricked us here. This changes belongs to the method onStageCompleted (and definitely not for onExecutorUnblacklisted). This is where I remove completed stages from the live executors.

Reply:
whoops, my fault!  thanks for explaining